<launch>
  <arg name="master_name" default="phantom"/>
  <arg name="slave_name" default="justin"/>
  <arg name="ref_frame" default="world"/>
  <arg name="rpy_offset" default="[0, 0, 0]"/>
  <arg name="rviz" default="true"/>

  
  <!-- Phantom node -->
  <node name="omni_state2" pkg="omni_common" type="omni_state2" output="screen"> 
    <param name="omni_name" type="string" value="$(arg master_name)"/>
    <param name="publish_rate" type="double" value="1000"/>
    <rosparam command="load" file="$(find rate_position_controller)/config/phantom_justin.yaml" />
  </node>
  
  <!-- Rate position controller -->
  <node name="rate_position_controller" pkg="rate_position_controller" type="rate_position_controller_good.py" output="screen">
    <param name="publish_rate" type="double" value="1000"/>
    <param name="slave_name" type="string" value="$(arg slave_name)"/>
    <param name="master_name" type="string" value="$(arg master_name)"/>
    <param name="reference_frame" type="string" value="$(arg ref_frame)"/>
    <rosparam param="rpy_offset" subst_value="True">$(arg rpy_offset)</rosparam>
<!--
    <remap from="$(arg slave_name)/state" to="/grips/endpoint_state"/>
-->
    <rosparam command="load" file="$(find rate_position_controller)/config/phantom_justin.yaml" />
  </node>
  
  <!-- Force handler node -->
  <node name="force_handler" pkg="rate_position_controller" type="force_handler.py" output="screen"> 
    <param name="publish_rate" type="double" value="1000"/>
    <param name="slave_name" type="string" value="$(arg slave_name)"/>
    <param name="master_name" type="string" value="$(arg master_name)"/>
    <param name="reference_frame" type="string" value="$(arg ref_frame)"/>
<!--
    <remap from="$(arg slave_name)/state" to="/grips/endpoint_state"/>
-->
    <rosparam command="load" file="$(find rate_position_controller)/config/phantom_justin.yaml" />
  </node>
  
  <!-- Obstacle collision node -->
<!--
  <node name="virtual_obstacles" pkg="rate_position_controller" type="virtual_obstacles.py" output="screen"> 
  </node>
-->
  <include file="$(find rate_position_controller)/launch/justin_receive_pos.launch">
  </include>
  
  <!-- Show in Rviz -->
  <group if="$(arg rviz)">
    <node name="rviz" pkg="rviz" type="rviz" output="screen" args="-d $(find rate_position_controller)/launch/phantom_justin.rviz"/>
<!--
<node name="rviz" pkg="rviz" type="rviz" output="screen" args="-d $(find grips_teleop)/config/cartesian_controller.rviz"/>
-->
  </group>
</launch>
